pkg/ctrdtaskapi/update.go

```diff
@@ -0,0 +1,18 @@
+package ctrdtaskapi
+
+import (
+	"github.com/containerd/typeurl"
+)
+
+func init() {
+	typeurl.Register(&PolicyFragment{}, "github.com/Microsoft/hcsshim/pkg/ctrdtaskapi", "PolicyFragment")
+}
+
+type PolicyFragment struct {
+	// Fragment is used by containerd to pass additional security policy
+	// constraint fragments as part of shim task Update request.
+	Fragment string `json:"fragment,omitempty"`
+	// Annotations hold arbitrary additional information that can be used to
+	// (e.g.) provide more context about Fragment.
+	Annotations map[string]string `json:"annotations,omitempty"`
```

Comment:
Do we have an expectation of what this will be used for? Pretty sure the overall update request already takes annotations, so rather we not just add it onto everything for future proofing.

Reply:
We briefly discussed with @KenGordon about potentially sending a "media-type", so that the guest would be able to process different content, e.g., tar, raw bytes etc. But I agree that we can add it later if we ever have a need to extend supported fragment formats

Reply:
Is the fragment format today a raw string of Rego? Do we know of cases where that will need to change in the future?

Reply:
as far as I know yes, it's a raw string of Rego. @KenGordon or @matajoh can we get your input here?

Reply:
At this level the payload is a COSE_Sign1 document base64 encoded so we can check the signatures inside the UVM.

Reply:
oh, right. I should've known this 🤦‍♂️

Reply:
Would it be better to rename this field to reflect this? Or at least add a comment stating it's a COSE_Sign1 doc with base64 encoding?